### ISSUE TITLE
Default for sort order being set as '1'

### ISSUE THREAD

**Issue by cowleyk** [nio-blocks/mongo#36]
https://github.com/nio-blocks/mongo/blob/master/mongodb_base.py#L59

```
"sort": [
        {
            "direction": "1",
            "key": "sort_order"
        }
    ],
```
This default configuration causes this error;
```
Traceback (most recent call last):
  File "/Users/kcowley/workspace/nio/nio/properties/util/object_type.py", line 28, in deserialize
    sub_instance.validate_dict(value)
  File "/Users/kcowley/workspace/nio/nio/properties/holder.py", line 85, in validate_dict
    prop.deserialize(value)
  File "/Users/kcowley/workspace/nio/nio/properties/base.py", line 145, in deserialize
    return self.type.deserialize(value, **merged_kwargs)
  File "/Users/kcowley/workspace/nio/nio/types/select.py", line 31, in deserialize
    value, kwargs.get('enum')))
TypeError: 1 does not match enum type: <enum 'SortDirection'>
```
Once the ascending/descending drop down is changed, the '1' is overwritten to the appropriate "ASCENDING" or "DESCENDING"

**Comment by hansmosh** [comment]
tldr: workaround is to set your sort order instead of using the default value.

`direction` in as enum type. It either needs to be a string equal to enum name or equal to the value. Valid values in this case would be `"ASCENDING"`, `"DESCENDING"`, `1` or `-1`.

It may be that in certain versions of nio, your direction was being converted to a string. I'm able to reproduce this issue with:
```  "nio": {
    "binary": "nio_full",
    "version": "2.0.0rc3",
    "build": "20170130"
  }
```

With the this version. , it works ok:
```  "nio": {
    "build": "20161219",
    "version": "2.0.0rc2",
    "binary": "nio_full"
  },
```

I can also reproduce the bug in the latest release. I'll file a bug report with our dev team.
```  "nio": {
    "binary": "nio_full",
    "build": "20170509",
    "version": "2.1.0"
  }
```

**Comment by hansmosh** [comment]
I've found that a workaround in the block code is to do
``` class Sort(PropertyHolder):
     key = StringProperty(title="Key", default="key")
     direction = SelectProperty(SortDirection,
-                               default=SortDirection.ASCENDING,
+                               default=SortDirection.ASCENDING.name,
                                title="Direction")```
